server/sandbox/create.go

```diff
@@ -0,0 +1,101 @@
+package sandbox
+
+import (
+	"context"
+)
+
+// Creator is the interface which reflects necessary steps in sandbox
+// creation. All methods have to be called in order to successfully creator a
+// pod sandbox.
+// TODO: add the necessary arguments to the methods
+type Creator interface {
+	// GenerateName prepares the name and unique identifier of the pod sandbox.
+	// It has to be ensured that the sandbox name does not already exist.
+	GenerateName()
+
+	// PullPauseImage ensures that the pause image of the container exists
+	// before container creation.
+	PullPauseImage()
+
+	// CreateContainer builds the container from the provided names,
+	// identifiers and pause image.
+	CreateContainer()
+
+	// SetupOCISpec prepares the basic, host specific OCI configuration,
+	// like setting the process args for the pause image execution or DNS
+	// configuration.
+	SetupOCISpec()
```

Comment:
Hm, I am unsure this should be separate from CreateInfraContainer

Reply:
This is obsolete with the latest version of this PR.